trace.go

```diff
@@ -224,3 +224,39 @@ type TraceSetter interface {
 	Error
 	SetTraces(...Trace)
 }
+
+// NewAggregate creates a new aggregate instance from the specified
+// list of errors
+func NewAggregate(errs []error) Aggregate {
+	if len(errs) == 0 {
+		return nil
+	}
+	return aggregate(errs)
+}
+
+// Aggregate interface combines several errors into one error
+type Aggregate interface {
```

Comment:
that part I understand, but why not simply make this a struct without extra interface. where do you use the interface particularly